k8s-path-cleanup.tf

```diff
@@ -3,75 +3,75 @@
+  path_cleanup_name      = "path-cleanup"
+  path_cleanup_full_name = "${local.k8s_app_alias}-path-cleanup"
+  path_cleanup_labels = {
+    "app.kubernetes.io/name"       = "${local.k8s_app_alias}-path-cleanup"
+    "app.kubernetes.io/instance"   = "${local.k8s_app_alias}-path-cleanup"
+    "app.kubernetes.io/version"    = var.path_cleanup_docker_image_version
     "app.kubernetes.io/managed-by" = local.k8s_app_alias
   }
-  scheduler_label_name_instance = {
-    "app.kubernetes.io/name"     = "${local.k8s_app_alias}-path-scheduler"
-    "app.kubernetes.io/instance" = "${local.k8s_app_alias}-path-scheduler"
+  path_cleanup_label_name_instance = {
+    "app.kubernetes.io/name"     = "${local.k8s_app_alias}-path-cleanup"
+    "app.kubernetes.io/instance" = "${local.k8s_app_alias}-path-cleanup"
   }
 }
 
-resource "kubernetes_deployment" "beekeeper_scheduler" {
+resource "kubernetes_deployment" "beekeeper_path_cleanup" {
   count = var.instance_type == "k8s" ? 1 : 0
   metadata {
-    name      = local.scheduler_full_name
+    name      = local.path_cleanup_full_name
     namespace = var.k8s_namespace
-    labels    = local.scheduler_labels
+    labels    = local.path_cleanup_labels
   }
 
   spec {
     # setting the number of replicas to greater than 1 is currently untested
     replicas = 1
     selector {
-      match_labels = local.scheduler_label_name_instance
+      match_labels = local.path_cleanup_label_name_instance
     }
 
     template {
       metadata {
-        labels = local.scheduler_label_name_instance
+        labels = local.path_cleanup_label_name_instance
         annotations = {
-          "iam.amazonaws.com/role" = aws_iam_role.beekeeper_k8s_role_scheduler_iam[count.index].arn
+          "iam.amazonaws.com/role" = aws_iam_role.beekeeper_k8s_role_path_cleanup_iam[count.index].arn
           "prometheus.io/scrape": var.prometheus_enabled
-          "prometheus.io/port": var.k8s_scheduler_port
+          "prometheus.io/port": var.k8s_path_cleanup_port
           "prometheus.io/path": "/actuator/prometheus"
         }
       }
 
       spec {
         container {
-          name              = local.scheduler_full_name
-          image             = "${var.path_scheduler_docker_image}:${var.path_scheduler_docker_image_version}"
+          name              = local.path_cleanup_full_name
+          image             = "${var.path_cleanup_docker_image}:${var.path_cleanup_docker_image_version}"
           image_pull_policy = var.k8s_image_pull_policy
 
           port {
-            name           = local.scheduler_name
-            container_port = var.k8s_scheduler_port
+            name           = local.path_cleanup_name
+            container_port = var.k8s_path_cleanup_port
           }
 
           liveness_probe {
             http_get {
               path = "/actuator/health"
-              port = var.k8s_scheduler_port
+              port = local.path_cleanup_name
```

Comment:
Shouldn't this be `var.k8s_scheduler_port`?

Reply:
For some reason these changes arent being recognised properly - I edited the `k8s-cleanup.tf` file, renamed it to `k8s-path-cleanup.tf` and everywhere it said `cleanup` I changed it to `path-cleanup`. In the original file this line says `port = local.cleanup_name` https://github.com/ExpediaGroup/apiary-lifecycle/blob/master/k8s-cleanup.tf#L62 but you have a point, it doesnt look right does it?

Reply:
Hmm, in this case it may be right, since it refers to the port by name.

Reply:
Have you tried to access the endpoint? Does it behave correctly? If yes, the port is correct.

Reply:
Update: we have discussed this and its okay as the port is being referenced by name, see another example here https://github.com/kubernetes-csi/livenessprobe